x-pack/filebeat/input/azureeventhub/sanitization_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+//go:build !aix
+// +build !aix
+
+package azureeventhub
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+)
+
+func TestSanitizeMessage(t *testing.T) {
```

Comment:
While all these tests make sense, they bootstrap a lot of things, when the methods themselves are pretty simple.
How about testing the full-blown setup once, and then testing each individual method directly?
That would allow more fine-tuned testing of those sanitizations, and simplification of those tests.

Also, you could reduce repetition in those full-blown tests by using subtests: https://go.dev/blog/subtests